r-lib-error-1.0.md

```diff
@@ -0,0 +1,60 @@
+
+# Specification of r-lib errors, version 1.0
+
+Note: this is an experimental draft.
+
+Note: version 1.0 corresponds to the `rlang_error` class, as
+implemented in rlang 0.4.0.
+
+## Classes
+
+An r-lib error object always has at least classes `rlib_error`, `error`
+and `condition`. It may have other classes.
```

Comment:
Should we mention these should be the root super classes?

Reply:
Yeah, I was thinking about that as well, but I don't really know if it is important.

Reply:
I think we don't need to require this, even though it will be the most common case. Adding a new class that comes after `rlib_error` or even `error` does not seem to make too much sense, but maybe there is some use case that we overlook here.

Reply:
I think it should be UB to add super classes after `rlib_error` or `error`. You should only do this if the constructor allows it. Since that's a spec we don't have a constructor, so we should mention it.